core/src/main/resources/jenkins/model/Jenkins/404.jelly

```diff
@@ -0,0 +1,54 @@
+<!--
+The MIT License
+
+Copyright (c) 2021 Daniel Beck
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
+    <!--
+      This is the page designated by web.xml to handle 404 errors. Show a nice error message and suggest logging in if applicable.
+      Generally kept very similar to 'oops.jelly'.
+    -->
+    <st:statusCode value="${response.getStatus()}" />
+    ${request.session.setAttribute('from', request.getAttribute('javax.servlet.error.request_uri'))}
+    <l:layout title="Jenkins" type="one-column">
+        <l:header />
+        <l:main-panel>
+            <h1 style="text-align: center">
+                <img src="${imagesURL}/rage.png" height="179" width="154"/> <span style="font-size:50px"><st:nbsp/>${%Oops!}</span>
+            </h1>
+            <div id="error-description" style="text-align: center">
+                <h2>${%title}</h2>
+                <p>
+                    <j:choose>
+                        <j:when test="${app.useSecurity}">
```

Comment:
This is a bit annoying -- we don't know who the current user is, so we cannot do permission checks to show a more tailored message.